### PR TITLE
feat(deps): Update Android SDK Version to 34

### PR DESCRIPTION
**Pull request by trindadedev13**
As you know, this PR is not ready, because ```TileService#startActivityAndCollapse``` is deprecated in SDK 34 and above.

### CODE REVIEW

**Comment by WSTxda** [comment]
In api 34 `StartActivityAndCollapse` in MusicSearchQuickSettings is deprecated, need create another method to collapse qs painel after lauch intent.

**Comment by antsummer** [comment]
> In api 34 `StartActivityAndCollapse` in MusicSearchQuickSettings is deprecated, need create another method to collapse qs painel after lauch intent.

okay, I'll take a look later then

**Comment by antsummer** [comment]
@WSTxda please re-view PR

File: app/src/main/java/com/wstxda/gsl/MusicSearchQuickSettings.kt [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+            /* use PendingIntent for the SDK and above */
+            val pendingIntent = PendingIntent.getActivity(
+                this,
+                0,
+                Intent("com.google.android.googlequicksearchbox.MUSIC_SEARCH").apply {
+                    flags = Intent.FLAG_ACTIVITY_NEW_TASK
+                },
+                PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
             )
```
Comment (WSTxda) [inline app/src/main/java/com/wstxda/gsl/MusicSearchQuickSettings.kt:23]: This method is correct, although it cannot collapse system qs panel, apparently this behavior is now managed by the activity being opened or by the system itself. you could probably try some **workarounds** but honestly I feel like it's not worth it.

I was able to try the changes and the activity is also not invoked on HyperOS A14 for some reason (stupid probably)

Comment (antsummer) [reply app/src/main/java/com/wstxda/gsl/MusicSearchQuickSettings.kt:23]: > This method is correct, although it cannot collapse system qs panel, apparently this behavior is now managed by the activity being opened or by the system itself. you could probably try some **workarounds** but honestly I feel like it's not worth it.
> 
> I was able to try the changes and the activity is also not invoked on HyperOS A14 for some reason (stupid probably)

Hmm, unfortunately I don't have Android 14 to test.

Comment (WSTxda) [reply app/src/main/java/com/wstxda/gsl/MusicSearchQuickSettings.kt:23]: `code=START_ABORTED` apparently the system is blocking launch of the intent for security policy reason.

`BAL_BLOCK` suggests that the action was blocked by Activity Management,  which reinforces that somehow opening the intent is outside of security

`08-25 01:15:03.157  1062  2196 I ActivityTaskManager: START u0 {act=com.google.android.googlequicksearchbox.MUSIC_SEARCH flg=0x10000000 cmp=com.google.android.googlequicksearchbox/.MusicSearchGatewayInternal} with LAUNCH_MULTIPLE from uid 10031 pid -1 (BAL_BLOCK) result code=START_ABORTED`

Maybe that's why it doesn't initialize, maybe exploiting an internal API is the solution?

Comment (antsummer) [reply app/src/main/java/com/wstxda/gsl/MusicSearchQuickSettings.kt:23]: Well, I just don't understand one thing, in SDK 34, what happens, what would this collapse be?

Comment (WSTxda) [reply app/src/main/java/com/wstxda/gsl/MusicSearchQuickSettings.kt:23]: We currently use `StartActivityAndCollapse` to get the behavior of, when tapping, launch the intent and close the system quick settings panel. This is required to display the search and display the results in the browser.

Without this behavior the entire process above is done without closing the panel.

As of sdk 34 this is no longer allowed, and apparently there are security restrictions. For this reason i kept the project with sdk 33.

Comment (antsummer) [reply app/src/main/java/com/wstxda/gsl/MusicSearchQuickSettings.kt:23]: aaa, so when the tile is clicked the panel is not closed, now I understand.

Comment (WSTxda) [reply app/src/main/java/com/wstxda/gsl/MusicSearchQuickSettings.kt:23]: Exactly, and open intent is blocked for security reason

**Comment by WSTxda** [comment]
Ok, i made the decision to just update the java, then keep the sdk target to 33, due this reason https://github.com/WSTxda/Google-Shortcuts-Launcher/pull/6#discussion_r1730208283

**Comment by aquilesTrindade13** [comment]
> Ok, i made the decision to just update the java, then keep the sdk target to 33, due this reason [#6 (comment)](https://github.com/WSTxda/Google-Shortcuts-Launcher/pull/6#discussion_r1730208283)

OK, I won't close the PR, but I'll mark it as a draft. When I have time, I'll study it to see something that can replace this call without the depreciation, of course, that is, if you don't find it first.

**Pull request closed without merging**